src/Time/ChooseLtsStepSize.cpp

```diff
@@ -22,7 +22,10 @@ TimeDelta choose_lts_step_size(const Time& time, const double desired_step) {
   // n because of floating-point truncation.  The inner ceil call
   // avoids this problem.
   const auto desired_step_power =
-      static_cast<size_t>(std::ceil(std::log2(std::ceil(desired_step_count))));
+      desired_step == std::numeric_limits<double>::infinity()
```

Comment:
This needs an `abs()`, since the step can be negative, or check for `desired_step_count == 0.0` instead.